tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,133 @@
+
+    mock_ansible_module.fail_json.side_effect = mock_cvpClient.api.fail_json
+
+    cv_tools = CvDeviceTools(mock_cvpClient, mock_ansible_module, 'serialNumber')
+
+    return mock_ansible_module, mock__get_device, cv_tools, mock_get_container_current
+
+
+@pytest.mark.state_present
+class TestApplyBundle():
+    """
+    Contains unit tests for apply_bundle()
+    """
+    @pytest.mark.parametrize(
+        "expected",
+        [
+            (True),    #  success
+            (False),   # failure
+        ],
+    )
+    def test_apply_bundle(self, setup, expected):
+        """
+        Test when current_image_bundle and assigned_image_bundle are different
+
+        :param setup: fixture
+        :param expected: output expected from apply_bundle
+
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+        mock__get_device.return_value = cv_data
+        if not expected:
+            # removing node_id for failure test-case
+            image_bundle['id'] = None
+
+        result = cv_tools.apply_bundle(user_inventory=user_topology)
+        assert result[0].success == expected
+        assert result[0].changed == expected
+
+        # resetting imageBundle id
+        image_bundle['id'] = 'imagebundle_1658329041200536707'
+
+    def test_apply_bundle_same_image(self, setup):
+        """
+        Test when current_image_bundle and assigned_image_bundle are same
+        :param setup: fixture
+
+        if both image bundles are same, nothing to do.
+        """
+        user_topology = DeviceInventory(data=device_data)
+        _, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+
+        cv_data_same = deepcopy(cv_data)
+        cv_data_same['imageBundle']['bundleName'] = device_data[0]['imageBundle']
+        mock__get_device.return_value = cv_data_same
+
+        result = cv_tools.apply_bundle(user_inventory=user_topology)
+        assert result[0].success == False
+        assert result[0].changed == False
+
+    def test_apply_bundle_cvp_api_error(self, setup):
+        """
+        Test for CvpApiError.
+        :param setup: fixture
+
+        """
+        user_topology = DeviceInventory(data=device_data)
+        mock_ansible_module, mock__get_device, cv_tools, mock_get_container_current = setup
+        mock_get_container_current.return_value = current_container_info
+        mock__get_device.return_value = cv_data
+        image_bundle['id'] = 'error_id'
+
+        with pytest.raises(SystemExit) as pytest_error:
+            _ = cv_tools.apply_bundle(user_inventory=user_topology)
+        assert pytest_error.value.code == 1
+        expected_call = [call.fail_json(msg='Error applying bundle to device tp-avd-leaf2 : Image bundle ID is not valid')]
```

Comment:
```suggestion
        expected_call = [call.fail_json(msg='Error applying bundle to device tp-avd-leaf2: Image bundle ID is not valid')]
```
and change this line: https://github.com/aristanetworks/ansible-cvp/blob/f8b8e40c44dd44299b38a73409efc803281bcef9/ansible_collections/arista/cvp/plugins/module_utils/device_tools.py#L1593
to
``` str(catch_error)```